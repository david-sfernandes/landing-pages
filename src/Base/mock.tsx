/* eslint-disable import/no-anonymous-default-export */
import links from '../components/NavLinks/mock';
import gridMock from "../components/GridText/mock";
import GridText from "../components/GridText";

export default {
  children: (
    <>
      <GridText {...gridMock}/>
      <GridText {...gridMock} background/>
    </>
  ),
  links: links,
  logoData: {
    text: 'Logo',
    link: '#'
  },
  footerHtml: '<p>Footer test</p>',
}