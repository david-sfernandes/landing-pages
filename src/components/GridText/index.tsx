import Heading from "../Heading";
import SectionBackground from "../SectionBackground";
import TextComponent from "../TextComponent";
import Container, { Grid, GridElement } from "./style";

export default function GridText({
  title,
  description,
  grid,
  background = false,
}: GridTextProps) {
  return (
    <SectionBackground background={background}>
      <Container>
        <Heading size="xl" uppercase darkColor={!background} as="h2">
          {title}
        </Heading>
        <TextComponent>{description}</TextComponent>
        <Grid>
          {grid.map((el) => (
            <GridElement key={el.title}>
              <Heading size="md" darkColor={!background} as="h3">
                {el.title}
              </Heading>
              <TextComponent>{el.description}</TextComponent>
            </GridElement>
          ))}
        </Grid>
      </Container>
    </SectionBackground>
  );
}
