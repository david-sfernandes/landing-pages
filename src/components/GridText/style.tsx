import styled, { css } from "styled-components";
import * as Heading from "../Heading/style";
import * as TextComponent from "../TextComponent/style";

const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings['2xl']};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const GridElement = styled.div`
  ${Heading} {
    position: relative;
    left: 5rem;
  }
  ${Heading}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
  }
`;

export default Container;
