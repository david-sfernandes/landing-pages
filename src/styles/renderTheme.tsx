import { render } from "@testing-library/react";
import { ReactNode } from "react";
import { ThemeProvider } from 'styled-components';
import { theme } from "./theme";

export default function renderTheme(children: ReactNode) {
  return (
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
  )
}