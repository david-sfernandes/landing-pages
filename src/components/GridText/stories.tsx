import GridText from ".";
import mock from "./mock";

const story = {
  title: "GridText",
  component: GridText,
  args: mock,
};
export default story;

export const Template = (args: any) => <GridText {...args} />;
