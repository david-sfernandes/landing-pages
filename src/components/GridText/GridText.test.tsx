import GridText from '.';
import renderTheme from '../../styles/renderTheme';

import mock from './mock';

describe('<GridText />', () => {
  it('should render with background inactive', () => {
    const { container } = renderTheme(<GridText {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('should render with background active', () => {
    const { container } = renderTheme(
      <GridText {...mock} background/>,
    );
    expect(container).toMatchSnapshot();
  });
});