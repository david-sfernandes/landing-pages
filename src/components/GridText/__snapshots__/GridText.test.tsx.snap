// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridText /> should render with background active 1`] = `
.c4 {
  color: #ffffff;
  font-size: 56px;
  text-transform: uppercase;
}

.c9 {
  color: #ffffff;
  font-size: 32px;
  text-transform: none;
}

.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 32px;
}

.c0 {
  background: #ffffff;
  color: #0A1128;
  background: #0A1128;
  color: #ffffff;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 24px;
}

.c2 {
  default: .c5;
}

.c2 {
  margin-bottom: 48px;
}

.c7 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
}

.c8 {
  default: .c3;
  default: .c3;
}

.c8 {
  position: relative;
  left: 5rem;
}

.c8::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
}

@media (max-width:768px) {
  .c4 {
    font-size: 32px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
        >
          My grid
        </h2>
        <p
          class="c5 c6"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </p>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 1
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 2
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 3
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridText /> should render with background inactive 1`] = `
.c4 {
  color: #0A1128;
  font-size: 56px;
  text-transform: uppercase;
}

.c9 {
  color: #0A1128;
  font-size: 32px;
  text-transform: none;
}

.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 32px;
}

.c0 {
  background: #ffffff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 24px;
}

.c2 {
  default: .c5;
}

.c2 {
  margin-bottom: 48px;
}

.c7 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
}

.c8 {
  default: .c3;
  default: .c3;
}

.c8 {
  position: relative;
  left: 5rem;
}

.c8::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
}

@media (max-width:768px) {
  .c4 {
    font-size: 32px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
        >
          My grid
        </h2>
        <p
          class="c5 c6"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </p>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 1
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 2
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 3
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
